packages/svelte/tests/parser-modern/samples/snippets/output.json

```diff
@@ -28,7 +28,7 @@
 					{
 						"type": "Identifier",
 						"start": 43,
-						"end": 46,
+						"end": 54,
```

Comment:
notably, the identifier `end` position is now including the type annotation, not just the name itself.
Judging from the tests that's not a problem but I wanted to make sure to point this out in case someone thinks this could have bad consequences

Reply:
It's not just identifiers, it's any `Pattern` node. Turns out `@typescript-eslint/parser` does the same thing, though TypeScript keeps the node and the annotation separate

So it looks like there's no universally agreed-upon convention here. I don't see any real downside to doing it this way, so if it's hard to keep things as they were for whatever reason then I think this change is probably fine

Reply:
You're right, any `Pattern`, and yes `@typescript-eslint/parser` also does it that way (and in fact that one specced the TSEStree which the acorn plugin follows).

The only downside could be someone relying on this somewhere else, but since they switch the library, and since it was broken in many places before, too (which is why we needed that fix-walk) so it should be fine in practise.

I also checked language-tools, and turns out we already had to do "use type annotation start instead of node end because we only want the JS part" in a couple of places already, so no breakage there, either.

Reply:
I tend to keep it like it's now to adhere to the inofficial estree spec - this is good to merge if you agree